modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/GridNearReadRepairAbstractFuture.java

```diff
@@ -388,6 +398,58 @@ protected final void recordConsistencyViolation(
             strategy));
     }
 
+    /**
+     *
+     */
+    private Object sensitiveValue(boolean includeSensitive, EntryGetResult res,
+        Map<ByteArrayWrapper, Object> sensitiveValMap) {
+        if (res != null) {
+            CacheObject val = res.value();
+
+            try {
+                ByteArrayWrapper wrapped = new ByteArrayWrapper(val.valueBytes(ctx.cacheObjectContext()));
+
+                return sensitiveValMap.computeIfAbsent(wrapped, w ->
+                    includeSensitive ?
+                        ctx.unwrapBinaryIfNeeded(val, !deserializeBinary, false, null) :
+                        "[HIDDEN_VALUE#" + UUID.randomUUID() + "]");
```

Comment:
Why do we need random UUID here?

Reply:
UUID is just one of the possible choices here. 
As I told you above, this seems to be the best choice.
The main goal is to have different values different even after masking, but the same should still be the same.